scripts/commands/getstats.lua

```diff
@@ -25,59 +33,121 @@ commandObj.onTrigger = function(player)
-    player:printToPlayer(string.format('RAccuracy Base: %i ', target:getMod(xi.mod.RACC)), xi.msg.channel.SYSTEM_3)
-    player:printToPlayer(string.format('Total RAccuracy: %i ', target:getStat(xi.mod.RACC)), xi.msg.channel.SYSTEM_3)
-    player:printToPlayer(string.format('Ranged Weap Dmg: %i ', target:getRangedDmg()), xi.msg.channel.SYSTEM_3)
-    player:printToPlayer(string.format('EVA Base: %i ', target:getMod(xi.mod.EVA)), xi.msg.channel.SYSTEM_3)
-    player:printToPlayer(string.format('EVA Total: %i ', target:getStat(xi.mod.EVA)), xi.msg.channel.SYSTEM_3)
-    player:printToPlayer(string.format('Magic EVA Base: %i ', target:getMod(xi.mod.MEVA)), xi.msg.channel.SYSTEM_3)
-    player:printToPlayer(string.format('Attack Base: %i ', target:getMod(xi.mod.ATT)), xi.msg.channel.SYSTEM_3)
-    player:printToPlayer(string.format('Total Attack: %i ', target:getStat(xi.mod.ATT)), xi.msg.channel.SYSTEM_3)
-    player:printToPlayer(string.format('Defense Base: %i ', target:getMod(xi.mod.DEF)), xi.msg.channel.SYSTEM_3)
-    player:printToPlayer(string.format('Total Defense: %i ', target:getStat(xi.mod.DEF)), xi.msg.channel.SYSTEM_3)
-    player:printToPlayer(string.format('Magic Attack bonus: %i ', target:getMod(xi.mod.MATT)), xi.msg.channel.SYSTEM_3)
-    player:printToPlayer(string.format('Magic Defense bonus: %i ', target:getMod(xi.mod.MDEF)), xi.msg.channel.SYSTEM_3)
-    player:printToPlayer(string.format('Magic Accuracy bonus: %i ', target:getMod(xi.mod.MACC)), xi.msg.channel.SYSTEM_3)
-    -- player:printToPlayer(string.format('Total Subtle Blow: %i ', target:getMod(xi.mod.SUBTLE_BLOW)), xi.msg.channel.SYSTEM_3)
-    -- player:printToPlayer(string.format('Total Store TP: %i ', target:getMod(xi.mod.STORETP)), xi.msg.channel.SYSTEM_3)
-    -- player:printToPlayer('Cannot easily and accurately get Magic Evasion with current methods.')
+    if not options[option] then
+        local printString = 'Please choose a report type:'
+        for k, v in ipairs(options) do
+            printString = printString .. ' ' .. k .. '-' .. v
+        end
+
+        player:printToPlayer(printString)
+        return
+    end
+
+    local messages =
+    {
+        'Stats for ' .. target:getName()
+    }
+    if options[option] == 'base' then
+        table.insert(messages, string.format('MainJob(jID: %s) LV: %i / SubJob(jID: %s) LV: %i ',
+            target:getMainJob(), target:getMainLvl(), target:getSubJob(), target:getSubLvl()))
+
+        table.insert(messages, string.format('HP: %i/%i  MP: %i/%i (current/max) ',
+            target:getHP(), target:getMaxHP(), target:getMP(), target:getMaxMP()))
+
+        table.insert(messages, string.format('Total STR: %i ', target:getStat(xi.mod.STR)))
+        table.insert(messages, string.format('Total DEX: %i ', target:getStat(xi.mod.DEX)))
+        table.insert(messages, string.format('Total VIT: %i ', target:getStat(xi.mod.VIT)))
```

Comment:
I am not a fan of all this table inserting just to spit out prints while still using the if/else anyway.

Reply:
seemed cleaner than a million printToPlayer calls

Reply:
suggested restructure:
- string param
- table the type instead of the prints (so reverse your table param table: `["string"] = number` so you can `get[1]` it)
- on matching conditions, do prints for each of the type print as normal
- but on anything that has to be specific to an entity type, wrap in a check for that entity type. no more `elseif`
- print example of command use and optional params when used without params or invalid params. and return out

Reply:
switch statement also acceptable to kill if/else-ness